pkg/catalog/catalog.go

```diff
@@ -504,6 +504,37 @@ func (c *Catalog) GetRepositoryMetadata(ctx context.Context, repository string)
 	return c.Store.GetRepositoryMetadata(ctx, repositoryID)
 }
 
+// SetRepositoryMetadata set repository metadata
+func (c *Catalog) UpdateRepositoryMetadata(ctx context.Context, repository string, metadata graveler.RepositoryMetadata) error {
```

Comment:
- Doc code is not aligned with the func name
- The lower call probably can't check it - if metadata len is 0 just return (relevant also for delete)